practical/DeepClustering/task1_Yu_Zhou.py

```diff
@@ -0,0 +1,38 @@
+import torch
+
+class MiniBatchKMeans:
+    def __init__(self, n_clusters: int, batch_size: int, max_iter: int):
+        self.n_clusters = n_clusters
+        self.batch_size = batch_size
+        self.max_iter = max_iter
+        self.centroids = None
+
+    def fit(self, X):
+        # Initialize centroids randomly from the data points
+        rand_indices = torch.randint(0, X.shape[0], (self.n_clusters,))
+        self.centroids = X[rand_indices]
+
+        for _ in range(self.max_iter):
+            # Find a random batch
+            batch_indices = torch.randint(0, X.shape[0], (self.batch_size,))
```

Comment:
This is fine, but since you never reuse `batch_indices` you could simply throw this statement into the batch selection (next line).